components/datetime/src/mock/time_zone.rs

```diff
@@ -34,6 +34,7 @@ use core::str::FromStr;
 ///     .expect("Failed to parse a time zone.");
 /// ```
 #[derive(Debug, Default)]
+#[allow(clippy::exhaustive_structs)] // this type is stable
```

Comment:
Question: Why is this struct stable but the others aren't? I don't think I have enough context here. Could you please explain your decision for what gets marked as exhaustive and what doesn't?

Reply:
note: when I say "stable" here, I mean "stable post 1.0"

I feel like the mock stuff should probably be private post 1.0, actually.

The decision is mostly guesswork, alongside these rules of thumb:

 - inputs to ICU4X APIs (options, etc) are non_exhaustive
 - outputs are _probably_ exhaustive unless we think we will want to add fields
 - provider stuff is always exhaustive/stable since we _serialize_ stably